sdk/servicebus/service-bus/src/core/messageSender.ts

```diff
@@ -545,22 +547,22 @@ export class MessageSender extends LinkEntity {
         );
       }
       if (shouldReopen) {
-        await defaultLock.acquire(this.senderLock, () => {
```

Comment:
Interestingly, this acquiring of lock being outside of each retry resulted in any send operation having to wait for 3 hours in a worse case scenario to have the error on disconnected network bubble up. See https://github.com/Azure/azure-sdk-for-js/issues/4182#issuecomment-613226742

Reply:
But say the network did come back up in these 3 hours, then before this change, the queued up send operations would get to continue. But, with this change, these operations fail right away as the init() call inside send() is not governed by retries.

Reply:
I would recommend that we pull the lock + init() calls from `send()` and `sendBatch()` to inside `trySend()` such that the init() call is governed by the retry logic we have. This way, we will have a consistent experience where network issues will result in the send operation being retried appropriately and the error bubbling up to the user after the 3 attempts are exhausted.

Reply:
I'll take a look at that today.

I did think it was odd that the _entirety_ of the retry() was "locked" so this is worth the investigation.

Reply:
To clarify my above comment, I am suggesting that we move the below from `send()` and `sendBatch()` inside `trySend()` so that the act of initializing the links is governed by the retries we already have
```
if (!this.isOpen()) { await this.open() }
```

Reply:
I did this but slightly differently (as usual ;)) - I just moved the isOpen() early exit check into .open() but it's the same idea and always do an await on this.open() in _trySend().

Reply:
By moving the early exit inside `open()` and ensuring there is always an await on it results in an extra round trip over the event loop for cases when the link is open :)

Reply:
(talked offline)

Going to leave this in place. FOR NOW.